### ISSUE TITLE
Redesign of Account/UnixAccount/RadiusAccount intermediates

### ISSUE THREAD

**Issue by Firstyear** [kanidm/kanidm#436]
These structs in IDM are constructed from entries (at probably non-trivial cost), then used and thrown away. Not all elements would be used either. 

As well there are lot of duplications between these, so we could base these from account, and structure these up. Perhaps trait impls on entry or similar?

**Comment by CEbbinghaus** [comment]
Its been a lil while since I did the last Refactor. Keen on finishing this up by moving the accounts but unsure what path you want to take. 

Personally I'm happy with how the Groups ended up and would be happy to tackle the accounts in the same way. However there are more shared properties within Accounts and given that we cannot inherit (the normal way at least) it would likely require a lot of nesting or duplication (e.g RadiusAccount & ServiceAccount share the `valid_from` member).

The alternative (which we could still add to the existing design) would be a mass of Traits that each define a subset of properties accessible through methods. So for example a `ValidFrom` trait would define `valid_from() -> Option<OffsetDateTime>` implemented on different classes.

**Comment by Firstyear** [comment]
Honestly, I'd rather see them become a reference to an Entry, which just borrows and validates the needed parts of the entry. That way we can divide these up by usage a bit better like "AuthenticatingAccount" which has what need for auth, RadiusAccount which has just what's needed for radius etc. We should probably try to avoid clones here too for these. 

But there are some challenges with it and the need to resolve groups, I think some more thought needed. Mostly I really want to get away from so many macros/clones in these.

**Comment by CEbbinghaus** [comment]
Okay. Let's make part of the goal reducing the number of clones. Happy to revisit the groups and improve them as part of this design. I think a wrapped entity might be a good option. I'll give it a bit more of a detailed thought but "reduce clones" is a good guiding principle

**Comment by Firstyear** [comment]
Yeah, the original reason for this was to "validate the structure of an entry and then make it accessible" so that we didn't have to be doing option checks repeatedly (think getting credentials or groups etc). But yeah, it ended up with lots of clones and it's own mess. So I think a way to validate an entry has "some defined shape", and do it with less clones would be good.
